core/store/src/cold_storage.rs

```diff
@@ -173,6 +173,16 @@ fn get_keys_from_store(
                     }
                     keys
                 }
+                // TODO: write StateChanges values to colddb directly, not to cache.
+                DBKeyType::TrieKey => {
+                    let mut keys = vec![];
+                    store.iter_prefix_with_callback(
+                        DBCol::StateChanges,
+                        &block_hash_key,
+                        |full_key| keys.push(full_key[block_hash_key.len()..].to_vec()),
```

Comment:
I wonder if it’s worth trying to avoid allocation here though it does result in a bit more convoluted code.  Something like

```suggestion
                        |full_key| {
                            let mut full_key = Vec::from(full_key);
                            full_key.drain(..block_hash_key.len());
                            keys.push(full_key);
                        }
```